integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/TableContentChangeDatasetBuilder.java

```diff
@@ -80,9 +83,13 @@ public List<OpenLineage.OutputDataset> apply(LogicalPlan x) {
                   null));
     }
 
+    DataSourceV2Relation returnTable =
```

Comment:
`TableContentChangeDatasetBuilder` is an output dataset builder while `DataSourceV2ScanRelation` seems to be relation of input datasets as it does not make much sense to scan outputs. 

I think it would have been beneficial to provide some integration test to make sure we do not mix input & output datasets.